processor/tasks/CROWNBase.py

```diff
@@ -0,0 +1,315 @@
+import law
+import luigi
+import os
+import yaml
+import shutil
+from framework import console
+from law.config import Config
+from framework import HTCondorWorkflow, Task
+from law.task.base import WrapperTask
+from rich.table import Table
+from helpers.helpers import *
+import ast
+
+# import timeout_decorator
+import time
+
+
+class ProduceBase(WrapperTask):
+    """
+    collective task to trigger friend production for a list of samples,
+    if the samples are not already present, trigger ntuple production first
+    """
+
+    sample_list = luigi.Parameter()
+    analysis = luigi.Parameter()
+    config = luigi.Parameter()
+    dataset_database = luigi.Parameter(significant=False)
+    production_tag = luigi.Parameter()
+    shifts = luigi.Parameter()
+    scopes = luigi.Parameter()
+
+    def parse_samplelist(self, sample_list):
+        """
+        The function `parse_samplelist` takes a sample list as input and returns a list of samples, handling
+        different input formats.
+
+        :param sample_list: The `sample_list` parameter is the input that the function takes. It can be
+        either a string, a list of strings, or a file path pointing to a text file
+        :return: a list of samples.
+        """
+        if str(sample_list).endswith(".txt"):
+            with open(str(sample_list)) as file:
+                samples = [nick.replace("\n", "") for nick in file.readlines()]
+        elif "," in str(sample_list):
+            samples = str(sample_list).split(",")
+        else:
+            samples = [sample_list]
+        return samples
+
+    def sanitize_scopes(self):
+        """
+        The function sanitizes the scopes information by converting it to a list if it is a string or
+        leaving it unchanged if it is already a list.
+        """
+        # sanitize the scopes information
+        try:
+            self.scopes = ast.literal_eval(str(self.scopes))
+        except:
+            self.scopes = self.scopes
+        if isinstance(self.scopes, str):
+            self.scopes = self.scopes.split(",")
+        elif isinstance(self.scopes, list):
+            self.scopes = self.scopes
+
+    def sanitize_shifts(self):
+        """
+        The function sanitizes the shifts information by converting it to a list if possible and handling
+        any exceptions.
+        """
+        # sanitize the shifts information
+        try:
+            self.shifts = ast.literal_eval(str(self.shifts))
+        except:
```

Comment:
Wouldn't it be beneficial to state the expected errors explicitly/use contextlib.suppress if no change is applied in case of occurring Exception?

Reply:
s.o.